aggregator/src/datastore.rs

```diff
@@ -2775,10 +2839,101 @@ pub mod models {
     use std::{fmt::Display, ops::RangeInclusive};
     use uuid::Uuid;
 
-    // We have to manually implement [Partial]Eq for a number of types because the dervied
+    // We have to manually implement [Partial]Eq for a number of types because the derived
     // implementations don't play nice with generic fields, even if those fields are constrained to
     // themselves implement [Partial]Eq.
 
+    /// Represents a report as it is stored in the leader's database, corresponding to a row in
+    /// `client_reports`, where `leader_input_share` and `helper_encrypted_input_share` are required
+    /// to be populated.
+    #[derive(Clone, Derivative)]
+    #[derivative(Debug)]
+    pub(crate) struct LeaderStoredReport<const L: usize, A>
+    where
+        A: vdaf::Aggregator<L>,
+        A::InputShare: PartialEq,
+        A::PublicShare: PartialEq,
```

Comment:
Should we drop these constraints from the struct, and only state them on the PartialEq/Eq implementations, and anything else that needs them?